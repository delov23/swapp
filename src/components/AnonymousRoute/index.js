export { default as AnonymousRoute } from './AnonymousRoute';
