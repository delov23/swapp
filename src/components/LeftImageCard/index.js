export { default as LeftImageCard } from './LeftImageCard';
