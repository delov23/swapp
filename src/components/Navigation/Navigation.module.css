@import "../../css/colors.module.css";

.navigation {
  display: flex;
  justify-content: space-between;
  padding: 1rem;
}

.navigationLight {
  composes: navigation;
  background-color: mainBlack;
}

.navigationDark {
  composes: navigation;
  background-color: #333;
}

.siteLogo {
  font-family: Starwars;
  font-weight: 200;
  font-size: 2rem;
  color: mainYellow;
  cursor: pointer;
  background: none;
  border: none;
  outline: none;
}

.links {
  display: flex;
  align-items: center;
  padding: 0;
  margin: 0;
  list-style: none;
}

.linkContainer {
  margin-left: 0.5rem;
}

.link {
  color: mainCyan;
}

.logout {
  color: mainCyan;
  padding: 0.4rem;
  border-radius: 0.2rem;
  border-width: 1.5px;
  border-style: solid;
  -webkit-border-radius: 0.2rem;
  -moz-border-radius: 0.2rem;
  -ms-border-radius: 0.2rem;
  -o-border-radius: 0.2rem;
}

.logoutLight {
  composes: logout;
  background: #eff2f7;
  border-color: lightBorder;
}

.logoutDark {
  composes: logout;
  background: #333;
  border-color: #3c4858;
}

@media screen and (max-width: 400px) {
  .siteLogo {
    font-size: 1.5rem;
  }
}