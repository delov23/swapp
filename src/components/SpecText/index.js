export { default as SpecText } from './SpecText';
