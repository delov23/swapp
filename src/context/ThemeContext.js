import { createContext } from 'react';

export const LIGHT_THEME = 'light';
export const DARK_THEME = 'dark';

const ThemeContext = createContext(LIGHT_THEME);

export default ThemeContext;
