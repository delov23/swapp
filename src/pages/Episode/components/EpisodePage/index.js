export { default as EpisodePage } from './EpisodePage';
