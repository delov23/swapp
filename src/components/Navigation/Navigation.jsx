import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt } from '@fortawesome/free-solid-svg-icons';

import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';
import styles from './Navigation.module.css';

const Navigation = ({ toggleTheme }) => {
  const theme = useContext(ThemeContext);

  return (
    <nav className={styles[themify('navigation', theme)]}>
      <button className={styles.siteLogo} onClick={toggleTheme}>
        SWApp
      </button>
      <ul className={styles.links}>
        <li>
          <Link to={'/episodes'} className={styles.link}>
            Episodes
          </Link>
        </li>
        <li className={styles.linkContainer}>
          <Link to={'/characters'} className={styles.link}>
            Characters
          </Link>
        </li>
        <li className={styles.linkContainer}>
          <Link to={'/logout'} className={styles[themify('logout', theme)]}>
            <FontAwesomeIcon icon={faSignOutAlt} />
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
