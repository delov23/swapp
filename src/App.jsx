import React, { useState } from 'react';
import { BrowserRouter, Switch, Redirect } from 'react-router-dom';
import { ApolloProvider } from '@apollo/react-hooks';

import client from './client';
import ThemeContext, { LIGHT_THEME, DARK_THEME } from './context/ThemeContext';
import { themify } from './utils/themify';
import styles from './App.module.css';

import Navigation from './components/Navigation/Navigation';
import { Login } from './pages/Login/';
import { AnonymousRoute } from './components/AnonymousRoute';
import { AuthRoute } from './components/AuthRoute';
import { Logout } from './pages/Logout';
import { Episodes } from './pages/Episodes';

const App = () => {
  const [theme, setTheme] = useState(
    localStorage.getItem('theme') || LIGHT_THEME,
  );

  const toggleTheme = () => {
    const newTheme = theme === LIGHT_THEME ? DARK_THEME : LIGHT_THEME;
    setTheme(newTheme);
    localStorage.setItem('theme', newTheme);
  };

  const handleHome = () => <Redirect to="/episodes" />;

  return (
    <ApolloProvider client={client}>
      <ThemeContext.Provider value={theme}>
        <div className={styles[themify('appWrapper', theme)]}>
          <BrowserRouter>
            <Navigation toggleTheme={toggleTheme} />
            <Switch>
              <AuthRoute path="/" exact render={handleHome} />
              <AnonymousRoute path="/login" component={Login} />
              <AuthRoute path="/logout" component={Logout} />
              <AuthRoute path="/episodes" component={Episodes} />
            </Switch>
          </BrowserRouter>
        </div>
      </ThemeContext.Provider>
    </ApolloProvider>
  );
};

export default App;
