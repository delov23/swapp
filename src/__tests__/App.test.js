import React from 'react';
import { shallow, mount } from 'enzyme';

import App from '../App';
import { DARK_THEME } from '../context/ThemeContext';
import { Navigation } from '../components/Navigation';

describe('<App />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle the theme', async () => {
    const wrapper = mount(<App />);

    wrapper
      .find(Navigation)
      .find('.siteLogo')
      .simulate('click');

    expect(localStorage.getItem('theme')).toEqual(DARK_THEME);
  });
});
