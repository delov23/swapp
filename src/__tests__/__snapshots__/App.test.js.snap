// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {
            "ROOT_QUERY": Object {
              "authenticated": false,
            },
          },
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {
            "ROOT_QUERY": Object {
              "authenticated": false,
            },
          },
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {
          Object {
            "definitions": Array [
              Object {
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "GeneratedClientQuery",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "authenticated",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
            "kind": "Document",
          } => Object {
            "definitions": Array [
              Object {
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "GeneratedClientQuery",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "authenticated",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
            "kind": "Document",
          },
        },
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {
              "ROOT_QUERY": Object {
                "authenticated": false,
              },
            },
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {
              "ROOT_QUERY": Object {
                "authenticated": false,
              },
            },
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {
            Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "authenticated",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            } => Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "authenticated",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            },
          },
          "watches": Set {},
        },
        "client": [Circular],
        "resolvers": Object {},
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {
                "ROOT_QUERY": Object {
                  "authenticated": false,
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {
                "ROOT_QUERY": Object {
                  "authenticated": false,
                },
              },
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {
              Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "authenticated",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              } => Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "authenticated",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              },
            },
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {
                "ROOT_QUERY": Object {
                  "authenticated": false,
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {
                "ROOT_QUERY": Object {
                  "authenticated": false,
                },
              },
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {
              Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "authenticated",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              } => Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "authenticated",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              },
            },
            "watches": Set {},
          },
          "client": [Circular],
          "resolvers": Object {},
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {
              "ROOT_QUERY": Object {
                "authenticated": false,
              },
            },
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {
              "ROOT_QUERY": Object {
                "authenticated": false,
              },
            },
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {
            Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "authenticated",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            } => Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "authenticated",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            },
          },
          "watches": Set {},
        },
      },
      "typeDefs": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "fields": Array [
              Object {
                "arguments": Array [],
                "directives": Array [],
                "kind": "FieldDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "authenticated",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Boolean",
                    },
                  },
                },
              },
            ],
            "interfaces": Array [],
            "kind": "ObjectTypeExtension",
            "name": Object {
              "kind": "Name",
              "value": "Query",
            },
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 49,
          "source": Object {
            "body": "extend type Query {
  authenticated: Boolean!
}",
            "locationOffset": Object {
              "column": 1,
              "line": 1,
            },
            "name": "GraphQL request",
          },
          "start": 0,
        },
      },
      "version": "2.6.3",
      "watchQuery": [Function],
    }
  }
>
  <ContextProvider
    value="light"
  >
    <div
      className="appWrapperLight"
    >
      <BrowserRouter>
        <Navigation
          toggleTheme={[Function]}
        />
        <Switch>
          <AnonymousRoute
            component={[Function]}
            path="/login"
          />
          <AuthRoute
            exact={true}
            path="/"
            render={[Function]}
          />
          <AuthRoute
            component={[Function]}
            path="/logout"
          />
          <AuthRoute
            component={[Function]}
            exact={true}
            path="/episodes"
          />
          <AuthRoute
            component={[Function]}
            path="/episodes/:episodeId"
          />
          <AuthRoute
            component={[Function]}
            exact={true}
            path="/characters"
          />
          <AuthRoute
            component={[Function]}
            path="/characters/:characterId"
          />
          <AuthRoute
            component={[Function]}
            path="/starships/:starshipId"
          />
        </Switch>
      </BrowserRouter>
    </div>
  </ContextProvider>
</ApolloProvider>
`;
